java-frontend/src/test/java/org/sonar/java/se/Expectations.java

```diff
@@ -235,109 +230,191 @@ public void visitTrivia(SyntaxTrivia syntaxTrivia) {
       collectExpectedIssues(syntaxTrivia.comment(), syntaxTrivia.startLine());
     }
 
-    private void collectExpectedIssues(String comment, int line) {
-      if (comment.startsWith(NONCOMPLIANT_COMMENT)) {
-        parseIssue(comment, line);
+    @VisibleForTesting
+    void collectExpectedIssues(String comment, int line) {
+      if (NONCOMPLIANT_COMMENT.matcher(comment).find()) {
+        IssueAndFlows issueAndFlows = parseIssue(comment, line);
+        if (issueAndFlows.parsed()) {
```

Comment:
Indeed, `parseIssue` always succeeds because we have at least the line. Thanks for catching this.